src/charm.py

```diff
@@ -174,7 +189,6 @@ def _install_vault_snap(self) -> None:
             vault_snap = snap_cache[VAULT_SNAP_NAME]
             if vault_snap.latest:
                 return
-            self.unit.status = MaintenanceStatus("Installing Vault")
```

Comment:
I think we should leave the Maintenance status. AFAIK, these are just used to signal progress within an event, and just signal to the juju user that something is happening. TBH we could probably make better use of it elsewhere, but I'd like to see this stay.

This might be an exception to the "status is only set in one place" paradigm, because these aren't attempting to set the final state of the unit.

Reply:
As it is, this wouldn't even work, update status runs at the end of every event and whatever's in there will overwrite anything else. If we want to keep maintenance status, we have to place it somewhere in collect status.

Reply:
I don't think that's true for `MaintenanceStatus`, it's a special case as far as I know.

While I couldn't find anything recent that explicitly states it, I did find [this](https://github.com/juju-solutions/layer-status?tab=readme-ov-file#conflict-resolution) that states

> Maintenance status is always set immediately

OK, maybe not the most reputable thing since it's 5 years old. But it also shows up in the docs such as [write your first charm](https://juju.is/docs/sdk/write-your-first-machine-charm)

```
    def _on_install(self, event: ops.InstallEvent):
        """Handle install event."""
        self.unit.status = ops.MaintenanceStatus("Installing microsample snap")
        os.system(f"snap install microsample --channel edge")
        self.unit.status = ops.ActiveStatus("Ready")  
```

And I've personally experienced it updating part-way through an event, unless I misunderstood something else.

Reply:
Huhh it seems you're right

Ref: https://github.com/canonical/operator/blob/main/ops/model.py#L345

Reply:
Fixed